File: src/main/java/com/cayena/backend/services/IProductService.java
package com.cayena.backend.services;

import com.cayena.backend.dtos.requesties.ProductRequest;
import com.cayena.backend.dtos.requesties.QuantityProductRequest;
import com.cayena.backend.dtos.responses.ProductAllResponse;
import com.cayena.backend.dtos.responses.ProductResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;

public interface IProductService {
    ProductResponse save(ProductRequest request);

    Page<ProductResponse> getAllProducts(PageRequest pageRequest);

    ProductAllResponse getProductById(Long id);

    void delete(Long id);

    ProductResponse update(Long id, ProductRequest request);

    ProductResponse updateQuantityStock(Long id, QuantityProductRequest request);
}

File: backend-kotlin/src/main/kotlin/com/cayena/backendkotlin/services/impl/IProductService.kt
package com.cayena.backendkotlin.services.impl

import com.cayena.backendkotlin.dtos.requests.ProductRequest
import com.cayena.backendkotlin.dtos.responses.ProductResponse

interface IProductService {
    fun createProduct( productRequest: ProductRequest): ProductResponse?

}
